### PR TITLE
Core-packages pages

### PR DESCRIPTION
**Pull request by xtuc**
Refers to #1021.

Index page:
core-packages-d1

@hzoo The index page could replace `usage > API` (https://babeljs.io/docs/usage/api/). We can then move the doc in `babel-core` (https://github.com/babel/babel/blob/master/packages/babel-core/README.md).

`usage > babel-register` (https://babeljs.io/docs/usage/babel-register/) same problem.

### CODE REVIEW

**Comment by xtuc** [comment]
@hzoo @existentialism What are your thought about this ? I will then move/link some pages from babel/babel.github.io to babel/babel.

File: docs/core-packages.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+layout: docs
+title: Babel's core packages
+description: The Babel repo is managed as a monorepo; it's composed of many npm packages
+permalink: /docs/core-packages/
+package: babel-core
+---
+
+## Other packages
+
+* [Types](types): Babel Types is a Lodash-esque utility library for AST nodes
```
Comment (hzoo) [inline docs/core-packages.md]: we can call it the package name if it's better - and does that link work?

Comment (xtuc) [inline docs/core-packages.md]: This is a relative link (points to `docs/core-packages/types`).

Comment (hzoo) [inline docs/core-packages.md]: Do you think it should be `babel-types` or `types`

Comment (xtuc) [inline docs/core-packages.md]: I did prefix all package in	c9a8804

**Comment by existentialism** [comment]
Might not need to solve this now, but some things to think about as we establish more consistency across docs:

- Title case for titles?
- Perhaps super nitpicky, but it's semi strange to have an h2 (Core) followed by another h2 (Install)

Others LGTM, nice work!

**Comment by xtuc** [comment]
It would be great to lint `babel/babel/packages/*/README.md`.

**Comment by hzoo** [comment]
yeah with a markdown AST plus custom rules 😛

**Comment by xtuc** [comment]
* /docs/usage/api is now `babel-core`README (ef667bbaa491d0b975af58d94f4858a6dcf887f9)
* /docs/usage/babel-register is now `babel-register` README (9fbb62b0489204f1c9848d4be28c544638f1f5fd)

@existentialism @hzoo

**Comment by hzoo** [comment]
Merging in https://github.com/babel/babel.github.io/pull/1044

**Pull request closed without merging**